src/S3/S3Client.php

```diff
@@ -324,7 +324,7 @@ public function listObjectsV2($input): ListObjectsV2Output
             $this->getEndpoint($input->requestUri(), $input->requestQuery())
         );
 
-        return new ListObjectsV2Output($response);
+        return new ListObjectsV2Output($response, $this->httpClient, $this, $input);
```

Comment:
HttpClient is not really needed, is it?

Reply:
Oh, yes. it is of course. It is used for the Steamable response. 

Sorry for the noise.